test/blackbox-tests/test-cases/no-name-field/run.t

```diff
@@ -17,3 +17,10 @@ executable(s) stanza works the same way
   File "dune", line 1, characters 0-36:
   Error: name field may not be omitted before dune version 1.1
   [1]
+
+there's only a public name but it's invalid as a name
+
+  $ dune build --root public-name-invalid-name
+  File "dune", line 1, characters 22-28:
+  Error: invalid library name
```

Comment:
This message doesn't make it clear how to resolve the issue. What about:

```
Error: invalid library name
Hint: library names must be non-empty and composed only of the following
characters: 'A'..'Z',  'a'..'z', '_'  or '0'..'9'. Public library names don't have this
restriction. You can either change this public name to be a valid library name
or add a `name` field with a valid library name.
```

Reply:
That's fine, but this error will also need to work for the case when the user enters a `name` field in a library stanza as well.